### PR TITLE
kpython2: Add new skeleton driver

### PR DESCRIPTION
**Pull request by 987123879113**
kpython and kpython2 share nothing in common besides the fact that they're PS2-based so I split kpython2 into its own driver. 

The HDD images were made by installing the game data onto a fresh HDD. All HDDs have been tested to make sure they boot properly and the ROM check in the operator menu returns OK (small exception: Thrill Drive 3 doesn't have a ROM check function but does have a data check during boot). However, because these are not HDD dumps straight from a real machine, I have marked them as `BAD_DUMP` so they can be replaced in the future if someone has the means to make proper dumps from a real machine. Likewise, the dongle files are generated using custom tools so they're marked as `BAD_DUMP` in case someone can replace them with real dongles dumps, but the dongle payload is understood and has been verified working against a real hardware. The `ps2_nvram` and `ps2_hdd_id` files *do* come from a real Python 2 machine but will need to be replaced if an HDD dump is replaced.

The BIOS dumped from a real Python 2 hash matches the `ps2-0190j-20030822.bin` "SCPH-50000 (Version 5.0 08/22/03 J)" BIOS already included in the `sony/ps2sony.cpp` driver so that file has been copied as-is. The dump is credited to Guru in the comments of `sony/ps2sony.cpp`, so credits to Guru for the BIOS.

Input bits have been roughly documented based on my research in emulating the Python 2 USB I/O board.

Hardware internals picture (PS2 + P2IO subboard): https://gamerepair.info/images/88_konami_python_2_io_board
Hardware internals picture (PS2 + P2IO subboard + MC-S PCB): https://forum.arcadeotaku.com/viewtopic.php?p=54202#p54202

New machines marked NOT_WORKING
-------------------------------
Dance Dance Revolution SuperNOVA (FDH:A:A:A:2006071300) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA (FDH:J:A:A:2006090600) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA (FDH:U:A:A:2006072400) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA 2 (GDJ:A:A:A:2007071100) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA 2 (GDJ:A:A:A:2007100800) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA 2 (GDJ:J:A:A:2007071100) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA 2 (GDJ:J:A:A:2007100800) [Taro, windyfairy, Shiz]
Dance Dance Revolution SuperNOVA 2 (GDJ:U:A:A:2007100800) [Taro, windyfairy, Shiz]
Dancing Stage SuperNOVA (FDH:E:A:A:2006032200) [Shiz, windyfairy]
Dancing Stage SuperNOVA (FDH:E:A:A:2006072500) [Shiz, windyfairy]
DrumMania V (E02:A:A:A:2005050200) [Taro, windyfairy, Shiz]
DrumMania V (E02:J:A:A:2005050200) [Taro, windyfairy, Shiz]
DrumMania V2 (F02:A:A:A:2005101600) [Taro, windyfairy, Shiz]
DrumMania V2 (F02:A:A:A:2005112800) [Taro, windyfairy, Shiz]
DrumMania V2 (F02:A:A:A:2006011201) [Taro, windyfairy, Shiz]
DrumMania V2 (F02:J:A:A:2005101600) [Taro, windyfairy, Shiz]
DrumMania V2 (F02:J:A:A:2005112800) [Taro, windyfairy, Shiz]
DrumMania V2 (F02:J:A:A:2006011201) [Taro, windyfairy, Shiz]
DrumMania V3 (F32:A:A:A:2006072600) [Taro, windyfairy, Shiz]
DrumMania V3 (F32:A:A:A:2006101800) [Taro, windyfairy, Shiz]
DrumMania V3 (F32:J:A:A:2006072600) [Taro, windyfairy, Shiz]
DrumMania V3 (F32:J:A:A:2006101800) [Taro, windyfairy, Shiz]
Guitar Freaks V (E03:A:A:A:2005050200) [Taro, windyfairy, Shiz]
Guitar Freaks V (E03:J:A:A:2005050200) [Taro, windyfairy, Shiz]
Guitar Freaks V2 (F03:A:A:A:2005101600) [Taro, windyfairy, Shiz]
Guitar Freaks V2 (F03:A:A:A:2005112800) [Taro, windyfairy, Shiz]
Guitar Freaks V2 (F03:A:A:A:2006011201) [Taro, windyfairy, Shiz]
Guitar Freaks V2 (F03:J:A:A:2005101600) [Taro, windyfairy, Shiz]
Guitar Freaks V2 (F03:J:A:A:2005112800) [Taro, windyfairy, Shiz]
Guitar Freaks V2 (F03:J:A:A:2006011201) [Taro, windyfairy, Shiz]
Guitar Freaks V3 (F33:A:A:A:2006072600) [Taro, windyfairy, Shiz]
Guitar Freaks V3 (F33:A:A:A:2006101800) [Taro, windyfairy, Shiz]
Guitar Freaks V3 (F33:J:A:A:2006072600) [Taro, windyfairy, Shiz]
Guitar Freaks V3 (F33:J:A:A:2006101800) [Taro, windyfairy, Shiz]
Thrill Drive 3 (D44:J:A:A:20050316) [Taro, windyfairy, Shiz]
Toy's March (E00:J:A:A:2005011602) [Taro, windyfairy, Shiz]
Toy's March 2 (F00:J:A:A:2005110400) [Taro, windyfairy, Shiz]

New software list additions
-------------------------------
Dance Dance Revolution SuperNOVA 2 Install Disk (JAA/AAA) [Taro]
Dance Dance Revolution SuperNOVA 2 Install Disk (UAA) [Ninevolt]
Dance Dance Revolution SuperNOVA Install Disk (UAA) [HybridStorm]
Dancing Stage SuperNOVA Install Disk (EAA) [Shiz]
DrumMania V2 (JAA/AAA) [Taro]
DrumMania V2 Version 1.01 (JAA/AAA) [Taro]
DrumMania V3 (JAA/AAA) [SakamotoNeko]
GuitarFreaks V2 (JAA/AAA) [Taro]
GuitarFreaks V2 Version 1.01 (JAA/AAA) [Taro]
GuitarFreaks V3 (JAA/AAA) [SakamotoNeko]

### CODE REVIEW

**Comment by Hydreigon223** [comment]
Mind adding in documentation the I/O board consisting on an H8S/2118 (not readable) and adding a placeholder dump?

**Comment by 987123879113** [comment]
I'm not adding anything I can't verify myself. If you can pinpoint it in a PCB pic then I'm willing to add it to the schematic but I don't see a reason to add placeholder dumps at this point. Someone can do that in the future when they start actually implementing things.

**Comment by Hydreigon223** [comment]
A friend of mine posted a pcb pic on a discord.
p2io

Sad it's kind of pointless to preserve these firmwares since HLE'ing its usb protocol is just as effective.

**Comment by 987123879113** [comment]
Added, thanks for the pic.

**Comment by AKuHAK** [comment]
Does anybody know how in Thrill Drive 3 two Python2 boards are connected together?

File: src/mame/konami/kpython2.cpp [hunk 1]
```diff
@@ -0,0 +1,1346 @@
...
+	m_maincpu->set_addrmap(AS_PROGRAM, &kpython2_state::ps2_map);
+
+	// Video hardware
+	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
+	screen.set_refresh_hz(60);
+	screen.set_screen_update(FUNC(kpython2_state::screen_update));
+	screen.set_size(640, 480);
+	screen.set_visarea(0, 639, 0, 479);
+
+	PALETTE(config, "palette").set_entries(65536);
```
Comment (cuavas) [inline src/mame/konami/kpython2.cpp:441]: You might want to use the devices from src/mame/sony/ps2sony.cpp as a starting point – it’s closer to being a real PlayStation 2 driver than anything else AFAIK.

File: src/mame/konami/kpython2.cpp [hunk 2]
```diff
@@ -0,0 +1,1346 @@
...
+
+	PORT_MODIFY("ANALOG2")
+    PORT_BIT(0x0000ffff, 0, IPT_PEDAL) PORT_SENSITIVITY(100) PORT_KEYDELTA(10) PORT_NAME("Accelerator Pedal")
+
+	PORT_MODIFY("ANALOG3")
+    PORT_BIT(0x0000ffff, 0, IPT_PEDAL) PORT_SENSITIVITY(100) PORT_KEYDELTA(10) PORT_NAME("Brake Pedal")
+
+    // TODO: Seatbelt toggle, returned by the BELT attachment device
+
+INPUT_PORTS_END
```
Comment (cuavas) [inline src/mame/konami/kpython2.cpp:556]: Looks like mixed tab/space indents snuck in here.

File: src/mame/konami/kpython2.cpp [hunk 3]
```diff
@@ -0,0 +1,1346 @@
...
+
+    // TODO: Drum pad inputs
+    // The drum pad inputs (P1 and P2 cymbal, drum L, and drum R) are pressure sensitive
+    // The drum pad values are returned by the drum pad attachment device
+
+INPUT_PORTS_END
+
+#define KPYTHON2_BIOS  \
+	    ROM_REGION32_LE(0x400000, "bios", 0) \
+		ROM_LOAD("ps2-0190j-20030822.bin", 0x000000, 0x400000, CRC(79d60546) SHA1(0ea98a25a32145dda514de2f0d4bfbbd806bd00c))
```
Comment (cuavas) [inline src/mame/konami/kpython2.cpp:577]: Same here, indents are messed up.  Just use tabs.

**Review by cuavas — COMMENTED** [review]
I believe it’s important to start documenting and preserving this stuff properly.  This is one of those annoying systems where every dump is different, with the encryption keyed to the specific machine (much like newer gambling systems).  It’s also important to preserve the installation media.

Given this is mostly a skeleton to hold the dumps, I’m not overly fussed on the machine configuration, but it might be better to base it on a more developed PlayStation 2 driver.

File: src/mame/konami/kpython2.cpp (hunk 1 above)
Comment (987123879113) [reply src/mame/konami/kpython2.cpp]: That's fair. I copypasted kpython.cpp as the base of this and made some small changes to remove the irrelevant code. I've redone it to use ps2sony.cpp as the base.

File: src/mame/konami/kpython2.cpp (hunk 2 above)
Comment (987123879113) [reply src/mame/konami/kpython2.cpp]: Went over the entire file and tried to fix any indentation issues I could find. Should be fixed now.

**Comment by Hydreigon223** [comment]
> I believe it’s important to start documenting and preserving this stuff properly. This is one of those annoying systems where every dump is different, with the encryption keyed to the specific machine (much like newer gambling systems). It’s also important to preserve the installation media.

I'm now curious to know which gambling systems have this type of encryption unless it isn't currently in Mame.

**Review by cuavas — APPROVED** [review]
I realise this contains copy/pasta – we can worry about devicifying the PlayStation 2 mainboard another time.

**Comment by AKuHAK** [comment]
@987123879113 is the `DANCE 86.4 FUNKY RADIO STATION (E01:J:A:A:2005040400`) also released on PythonII boards? It is missing from this pr.

**Pull request merged**